src/dftbp/math/symmetrise.F90

```diff
@@ -0,0 +1,131 @@
+!--------------------------------------------------------------------------------------------------!
+!  DFTB+: general package for performing fast atomistic simulations                                !
+!  Copyright (C) 2006 - 2025 DFTB+ developers group                                                !
+!                                                                                                  !
+!  See the LICENSE file for terms of usage and distribution.                                       !
+!--------------------------------------------------------------------------------------------------!
+
+!> Contains routines to symmetrize parts of a calculation
+module dftbp_math_symmetrize
```

Comment:
I am not sure about the naming. Those routines are really transport/contact specific (symmetrizing the first half of the atoms with the second half). Maybe we should call this `dftbp_*_contactsymm` and `TEquivContactAtoms`?

Reply:
In principle this could be generalized to other atom charge symmetries, but will rename for the moment.